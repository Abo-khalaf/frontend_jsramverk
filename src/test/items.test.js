import { render, fireEvent, screen, waitFor } from "@testing-library/react";
import Items from "../items";

import { mount, shallow } from "enzyme";
const editor = require("../editor.js").default;

describe("Filter function", () => {
  test("creat an item", () => {
    render(<Items />);
    expect(() => render(<Items />)).toBeTruthy();
    const createButton = screen.getByText("My Editor");
    expect(createButton).toHaveTextContent("My Editor");
  });

  // test("save test button", () => {
  //   const wrapper = shallow(<Items />);
  //   wrapper.find()
  //   expect(wrapper.state('postData')).toBe(0);

  //   // const createButton = screen.getByTestId("save-button");
  //   // fireEvent.click(createButton);
  //   // expect(wrapper.count).toBe(0);
  // });

  test("should show items", () => {
    const itemsWrapper = render(<Items />);
    expect(itemsWrapper).toMatchSnapshot();
  });

  test("should show items", () => {
    const itemsWrapper = render(<Items />);
    expect(itemsWrapper).toMatchSnapshot();

    const showButton = screen.getByTestId("show-button");
    fireEvent(
      showButton,
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true,
      })
    );
    expect(itemsWrapper).toMatchSnapshot();
  });
});
