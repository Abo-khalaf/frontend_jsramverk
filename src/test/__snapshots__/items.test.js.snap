// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter function should show items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <h2>
          My Editor
        </h2>
        <div>
          title:
           
          <input
            name="title"
            type="text"
            value=""
          />
        </div>
        <div
          class="Editor"
        >
          <div />
        </div>
        <button
          class="Button"
          data-testid="save-button"
        >
          save
        </button>
        <button
          class="Button"
        >
          Edit
        </button>
        <button
          class="Button"
        >
          Delete
        </button>
        <button
          class="Button"
          data-testid="show-button"
        >
          Show
        </button>
        <div>
          <ul />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <h2>
        My Editor
      </h2>
      <div>
        title:
         
        <input
          name="title"
          type="text"
          value=""
        />
      </div>
      <div
        class="Editor"
      >
        <div />
      </div>
      <button
        class="Button"
        data-testid="save-button"
      >
        save
      </button>
      <button
        class="Button"
      >
        Edit
      </button>
      <button
        class="Button"
      >
        Delete
      </button>
      <button
        class="Button"
        data-testid="show-button"
      >
        Show
      </button>
      <div>
        <ul />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Filter function should show items 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <h2>
          My Editor
        </h2>
        <div>
          title:
           
          <input
            name="title"
            type="text"
            value=""
          />
        </div>
        <div
          class="Editor"
        >
          <div />
        </div>
        <button
          class="Button"
          data-testid="save-button"
        >
          save
        </button>
        <button
          class="Button"
        >
          Edit
        </button>
        <button
          class="Button"
        >
          Delete
        </button>
        <button
          class="Button"
          data-testid="show-button"
        >
          Show
        </button>
        <div>
          <ul />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <h2>
        My Editor
      </h2>
      <div>
        title:
         
        <input
          name="title"
          type="text"
          value=""
        />
      </div>
      <div
        class="Editor"
      >
        <div />
      </div>
      <button
        class="Button"
        data-testid="save-button"
      >
        save
      </button>
      <button
        class="Button"
      >
        Edit
      </button>
      <button
        class="Button"
      >
        Delete
      </button>
      <button
        class="Button"
        data-testid="show-button"
      >
        Show
      </button>
      <div>
        <ul />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Filter function should show items 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <h2>
          My Editor
        </h2>
        <div>
          title:
           
          <input
            name="title"
            type="text"
            value=""
          />
        </div>
        <div
          class="Editor"
        >
          <div />
        </div>
        <button
          class="Button"
          data-testid="save-button"
        >
          save
        </button>
        <button
          class="Button"
        >
          Edit
        </button>
        <button
          class="Button"
        >
          Delete
        </button>
        <button
          class="Button"
          data-testid="show-button"
        >
          Show
        </button>
        <div>
          <ul />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <h2>
        My Editor
      </h2>
      <div>
        title:
         
        <input
          name="title"
          type="text"
          value=""
        />
      </div>
      <div
        class="Editor"
      >
        <div />
      </div>
      <button
        class="Button"
        data-testid="save-button"
      >
        save
      </button>
      <button
        class="Button"
      >
        Edit
      </button>
      <button
        class="Button"
      >
        Delete
      </button>
      <button
        class="Button"
        data-testid="show-button"
      >
        Show
      </button>
      <div>
        <ul />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
